src/Spice86.Core/Emulator/VM/Machine.cs

```diff
@@ -265,47 +268,48 @@ public class Machine : IDisposable {
         RegisterIoPortHandler(MidiDevice);
 
         // Services
-        CallbackHandler = new CallbackHandler(this, machineCreationOptions.LoggerService, MemoryMap.InterruptHandlersSegment);
+        CallbackHandler = new CallbackHandler(this, machineCreationOptions.LoggerService);
         Cpu.CallbackHandler = CallbackHandler;
+        InterruptInstaller = new InterruptInstaller(Memory, CallbackHandler, Cpu.FunctionHandler);
         
         VgaRom = new VgaRom();
         Memory.RegisterMapping(MemoryMap.VideoBiosSegment << 4, VgaRom.Size, VgaRom);
         VgaFunctions = new VgaFunctionality(Memory, IoPortDispatcher, BiosDataArea, VgaRom);
         VideoInt10Handler = new VgaBios(this, VgaFunctions, BiosDataArea, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(VideoInt10Handler);
         
         TimerInt8Handler = new TimerInt8Handler(this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(TimerInt8Handler);
         BiosKeyboardInt9Handler = new BiosKeyboardInt9Handler(this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(BiosKeyboardInt9Handler);
         
         BiosEquipmentDeterminationInt11Handler = new BiosEquipmentDeterminationInt11Handler(this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(BiosEquipmentDeterminationInt11Handler);
         SystemBiosInt15Handler = new SystemBiosInt15Handler(this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(SystemBiosInt15Handler);
         KeyboardInt16Handler = new KeyboardInt16Handler(
             this,
             machineCreationOptions.LoggerService,
             BiosKeyboardInt9Handler.BiosKeyboardBuffer);
-        RegisterCallbackHandler(KeyboardInt16Handler);
         SystemClockInt1AHandler = new SystemClockInt1AHandler(
             this,
             machineCreationOptions.LoggerService,
             TimerInt8Handler);
-        RegisterCallbackHandler(SystemClockInt1AHandler);
-
-        // Initialize DOS.
-        Dos = new Dos(this, machineCreationOptions.LoggerService);
-        Dos.Initialize(SoundBlaster, machineCreationOptions.Configuration);
 
         MouseDriver = new MouseDriver(Cpu, Memory, MouseDevice, machineCreationOptions.Gui, VgaFunctions, machineCreationOptions.LoggerService);
-        var mouseInt33Handler = new MouseInt33Handler(this, machineCreationOptions.LoggerService, MouseDriver);
-        RegisterCallbackHandler(mouseInt33Handler);
-        var mouseIrq12Handler = new BiosMouseInt74Handler(MouseDriver, DualPic, this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(mouseIrq12Handler);
-        var mouseCleanupHandler = new CustomMouseInt90Handler(MouseDriver, this, machineCreationOptions.LoggerService);
-        RegisterCallbackHandler(mouseCleanupHandler);
+        Dos = new Dos(this, machineCreationOptions.LoggerService);
 
+        if (Configuration.InitializeDOS is not false) {
```

Comment:
Same behaviour as before, but we need more granularity in configuration here

Reply:
This could be an option group:
https://github.com/commandlineparser/commandline/wiki/Verbs

for example:
Spice86 --initialize bios dos

Reply:
Created an issue to keep track of this, to do in a separate PR: https://github.com/OpenRakis/Spice86/issues/361